src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -136,11 +136,10 @@ public function start(): bool
         }
 
         $sessionId = $_COOKIE[session_name()] ?? null;
-        if ($sessionId && !preg_match('/^[a-zA-Z0-9,-]{22,}$/', $sessionId)) {
+        if ($sessionId && 'files' === $this->getSaveHandler()->getSaveHandlerName() && !preg_match('/^[a-zA-Z0-9,-]{22,}$/', $sessionId)) {
```

Comment:
`if ($sessionId && $this->saveHandler instanceof AbstractProxy && 'files' === $this->saveHandler->getSaveHandlerName() && !preg_match('/^[a-zA-Z0-9,-]{22,}$/', $sessionId)) {`

Reply:
Fixed. Given \Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage::setSaveHandler() I'm not sure how you could end up with something that is not an AbstractProxy but there's no harm in being safe and implementing according to the declared types.